File: android/app/src/main/java/com/studymode/app/study_mode_v2/StudyModeDeviceAdminReceiver.java
package com.studymode.app.study_mode_v2;

import android.app.admin.DeviceAdminReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

public class StudyModeDeviceAdminReceiver extends DeviceAdminReceiver {
    private static final String TAG = "StudyModeDeviceAdmin";

    @Override
    public void onEnabled(Context context, Intent intent) {
        super.onEnabled(context, intent);
        Log.d(TAG, "Study Mode Device Admin enabled");
    }

    @Override
    public void onDisabled(Context context, Intent intent) {
        super.onDisabled(context, intent);
        Log.d(TAG, "Study Mode Device Admin disabled");
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        super.onReceive(context, intent);
        String action = intent.getAction();
        Log.d(TAG, "Device Admin received action: " + action);
    }
}

File: android/app/src/main/kotlin/com/studymode/app/study_mode_v2/StudyModeDeviceAdminReceiver.kt
package com.studymode.app.study_mode_v2

import android.app.admin.DeviceAdminReceiver
import android.content.Context
import android.content.Intent
import android.widget.Toast

class StudyModeDeviceAdminReceiver : DeviceAdminReceiver() {
    
    override fun onEnabled(context: Context, intent: Intent) {
        super.onEnabled(context, intent)
        Toast.makeText(context, "Study Mode Device Admin enabled", Toast.LENGTH_SHORT).show()
    }

    override fun onDisabled(context: Context, intent: Intent) {
        super.onDisabled(context, intent)
        Toast.makeText(context, "Study Mode Device Admin disabled", Toast.LENGTH_SHORT).show()
    }
}